### ISSUE TITLE
Parsing error when parserOptions.ecmaVersion is not set

### ISSUE THREAD

**Issue by PaulTondeur** [mdx-js/eslint-mdx#366]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

latest

### Steps to reproduce

A very basic example following the instructions within this repo is already failing for me with a parsing error:
```
  0:0  error  Parsing error: ImportDeclaration should appear when the mode is ES6 and in the module context
```

I've reduced my `.eslintrc.json` to the simplest possible.

```json
{
  "plugins": ["@typescript-eslint"],
  "extends": [
    "next/core-web-vitals",
    "plugin:prettier/recommended",
    "plugin:mdx/recommended"
  ],
  "settings": {
    "mdx/code-blocks": true,
    "mdx/language-mapper": {}
  }
}
```

When I add the following to the `.eslintrc.json` it works as expected
```json
  "parserOptions": {
    "ecmaVersion": 12
  }
```

If this isn't a bug, it should be documented somewhere. It took a lot of time to figure this out.

### Expected behavior

My MDX file should be linted without errors

### Actual behavior

```0:0  error  Parsing error: ImportDeclaration should appear when the mode is ES6 and in the module context```

### Runtime

Node v16

### Package manager

yarn v2, yarn v1

### OS

macOS

### Build and bundle tools

Next.js

**Comment by JounQin** [comment]
Please provide a runnable reproduction.

There is a default `ecmaVersion` for parsing at https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-mdx/src/parser.ts#L48

**Comment by PaulTondeur** [comment]
What else would you like to have to reproduce? As detailed above with the basic eslint config. All that might be missing is another simplest mdx file with an import or jsx code.

**Comment by JounQin** [comment]
A demo GitHub repository for example.

**Comment by hisapy** [comment]
Hi everyone, 

In my case I ended up with the following `.eslintrc.json` config:

```json
{
  "plugins": ["@typescript-eslint"],
  "extends": [
    "next/core-web-vitals",
    "plugin:@typescript-eslint/recommended",
    "plugin:storybook/recommended",
    "prettier"
  ],
  "rules": {
    "@typescript-eslint/no-unused-vars": "error",
    "@typescript-eslint/no-explicit-any": "error",
    "semi": [1, "never"]
  },
  "overrides": [
    {
      "files": ["*.mdx", "*.md"],
      "extends": "plugin:mdx/recommended",
      "parserOptions": {
        "ecmaVersion": "latest"
      }
    }
  ]
}
```

What I added for mdx is in overrides.

I had to add *.md in files because `#` was being reported as invalid character in VSCode

**Comment by JounQin** [comment]
```sh
# yarn 1
yarn add https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-plugin-mdx
# yarn 2, 3
yarn add eslint-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-mdx/_pkg.tgz eslint-plugin-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-plugin-mdx/_pkg.tgz
# npm
npm i https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-plugin-mdx
```

Please help us to test the fresh new version.

**Comment by JounQin** [comment]
Close in favor of #402, wait for a release from `release-v1` branch @wooorm.

**Comment by sneko** [comment]
I can't explain why but this issue is still happening with latest version. I had to make a mix of multiple suggestions above:
```json
  overrides: [
    {
      files: ['*.md', '*.mdx'],
      extends: 'plugin:mdx/recommended',
      parserOptions: {
        ecmaVersion: 12,
      },
    },
  ],
```

(using `latest` for `ecmaVersion` was doing nothing, maybe that's the `13 (latest)` that is problematic?)
